contrib/coccinelle/zero_trace_reason.cocci

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0
+/// Prevent passing 0 as a reason to send_trace_notify(), because 0 is a valid
+/// value corresponding to new conntrack connections, and passing it instead of
+/// TRACE_REASON_UNKNOWN may confuse Hubble.
+// Confidence: Low
+// Copyright Authors of Cilium.
+// Comments:
+// Options: --include-headers
+
+@initialize:python@
+@@
+
+cnt = 0
+
+
+@rule@
+expression e1, e2, e3, e4, e5, e6, e7;
+position p;
+@@
+
+(
+- send_trace_notify(e1, e2, e3, e4, e5, e6, 0@p, e7);
++ send_trace_notify(e1, e2, e3, e4, e5, e6, TRACE_REASON_UNKNOWN, e7);
```

Comment:
Do we need the same for `update_trace_metrics`? Maybe we should also enforce this for the `struct trace_ctx` declarations?

Reply:
Hubble doesn't care about those metrics, so I think the impact of a wrong value will not affect Hubble (which is confused about wrong reasons in trace_notify), but agreed that we should probably also update the metrics just in case.

Reply:
Should be addressed now